platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ComponentReportingTasksPlugin.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.diagnostics.internal.DiagnosticsTaskNames;
+
+/**
+ * Plugin that adds tasks to report on the deprecated software model components configured for the project.
+ *
+ * @since 9.0
+ */
+@Incubating
+abstract public class ComponentReportingTasksPlugin implements Plugin<Project> {
```

Comment:
Does this plugin class need to be in the public API? Same question for the other plugin class.

Reply:
Well, the plugins that they were a part of (`HelpTasksPlugin` and `SoftwareReportingTasksPlugin`) are also public.  I just pulled these out into similar, different plugins, but I don't suppose they need to be public for any reason.  We're not exposing anything new, just moving it to a different plugin.  Having said that, I don't think they _need_ to be public.  I've made them internal.